### DISCUSSION TITLE
Perceptual Hash (again)

### DISCUSSION THREAD

**Discussion by f1-outsourcing** [ImageMagick/ImageMagick6#209] [General]
### ImageMagick version

6.9.12-70

### Operating system, version and so on

linux

### Description

I have been reading these pages[1] some of them are quite old, and I wonder if you still need to 'script' using this functionality.

eg. if I use
`identify -verbose -features 1 -moments -unique barn.jpg`
I am not getting these long binary strings that are on Fred's page

However when I use it on one of my random images like this
`identify -verbose -features 1 -moments -unique 3.jpg`

I am indeed getting these types of results
```
bmh 001011001000011000001000111...
bdh 1001000110111000011111100111101000111011...
avh 111101110000..
pfh 1000000....
bmh 0010110....
bdh 100100011011100...
avh 1111011100001..
pfh 10000000001111...
```

My questions would be

1. Are these binary strings directly outputted by identify sufficient for identifying the same image, or still scripts necessary that process the output of identify/convert?
2. Why do I not get these binary strings on the barn.jpg?
3. Is it even possible to generate with running identify on the barn.jpg the same output as on Fred's page? Or have the algorithms changed and this is not possible any more?
4. What would be a good way to test, eg a convert command to slightly modify the test image more and more until it is not possible to generate the same hash?

[1]
http://www.fmwconcepts.com/imagemagick/phashes/index.php
https://imagemagick.org/discourse-server/viewtopic.php?f=4&t=24906
https://im.snibgo.com/phashtest.htm

*I am totally not educated in image processing

**Comment by fmw42** [comment]
Those strings come from running my script phashes.  I do not see that you have run that. So why is your identify reporting them.  They are stored in the header “comments” of the image after running the script. They have no connection to either -features or -moments.

Have you run my script phashes?

Please explain in more detail what you have done before running identify -verbose

**Reply by f1-outsourcing** [reply]
> Those strings come from running my script phashes. I do not see that you have run that. So why is your identify reporting them. They are stored in the header “comments” of the image after running the script. They have no connection to either -features or -moments.
> 

Pfff, I feel like an idiot

> Have you run my script phashes?

Yes, but I thought they were not working on my older centos, I am getting this

```
[@svr ]$ phashes -m all  3.jpg
convert: delegate library support not built-in `./hashes_1_16806.mpc' (FFTW) @ warning/fourier.c/ForwardFourierTransformImage/640.
```
And on newer centos 9, I was also getting this

```
[@~]# phashes -m all  3.jpg
convert: delegate library support not built-in `./hashes_1_1515841.mpc' (FFTW) @ warning/fourier.c ForwardFourierTransformImage/1010.
```

**Reply by fmw42** [reply]
Polar FFT HASH requires FFTW delegate
